src/Model/Subsite.php

```diff
@@ -257,6 +257,10 @@ public static function getSubsiteIDForDomain($host = null, $checkPermissions = t
             $host = $_SERVER['HTTP_HOST'];
         }
 
+        // Remove ports, we aren't concerned with them in terms of detecting subsites via domains
+        $hostParts = explode(':', $host);
```

Comment:
`explode(':', $host, 2)`

Reply:
I'm not sure this is necessary. What case do you see it being of benefit? If someone specifies something invalid?
It could be pointed out that `user:pass@domain:port` has 2 colons I suppose...

Reply:
Well, it's faster if the internals know to stop scanning the string after the first one is found, and we use reset to get the first part later in the function so the invalid point is moot

Reply:
Patched in cd71c2b @dhensby